esinstall/src/index.ts

```diff
@@ -341,17 +335,218 @@ export async function install(
+    logLevel: 'info',
+    metafile,
+  });
+
+  await fs.promises.unlink(tsconfigTempFile);
+
+  const meta = JSON.parse(await (await fs.promises.readFile(metafile)).toString());
+
+  const importMap = metafileToImportMap({
+    installEntrypoints,
+    meta,
+    destLoc: destLoc,
+  });
+
+  const stats = metafileToStats({meta, destLoc});
+
+  return {stats, importMap};
+}
+
+function metafileToImportMap(_options: {
+  installEntrypoints: Record<string, string>;
+  meta: Metadata;
+  destLoc: string;
+}): ImportMap {
+  const {destLoc: destLoc, installEntrypoints: installEntrypoints, meta} = _options;
+  const inputFiles = Object.values(installEntrypoints).map((x) => path.resolve(x)); // TODO replace resolve with join in cwd
+  const inputFilesToSpecifiers = invert(installEntrypoints);
+
+  const importMaps: Record<string, string>[] = Object.keys(meta.outputs).map((output) => {
+    // chunks cannot be entrypoints
+    if (path.basename(output).startsWith('chunk.')) {
+      return {};
+    }
+    const inputs = Object.keys(meta.outputs[output].inputs).map((x) => path.resolve(x)); // TODO replace resolve with join in cwd
+    const input = inputs.find((x) => inputFiles.includes(x));
+    if (!input) {
+      return {};
+    }
+    const specifier = inputFilesToSpecifiers[input];
+    return {[specifier]: './' + toUnixPath(path.normalize(path.relative(destLoc, output)))};
+  });
+  const importMap = Object.assign({}, ...importMaps);
+  return {imports: importMap};
+}
+
+function metafileToStats(_options: {meta: Metadata; destLoc: string}): DependencyStatsOutput {
+  const {meta, destLoc} = _options;
+  const stats = Object.keys(meta.outputs).map((output) => {
+    const value = meta.outputs[output];
+    // const inputs = meta.outputs[output].bytes;
+    return {
+      path: output,
+      isCommon: ['chunk.'].some((x) => path.basename(output).startsWith(x)),
+      bytes: value.bytes,
+    };
+  });
+
+  function makeStatObject(value) {
+    const relativePath = toUnixPath(path.relative(destLoc, value.path));
+    return {
+      [relativePath]: {
+        size: value.bytes,
+        gzip: 0, // TODO do we want to waste time compressing to show stats?
+        brotly: 0,
+        delta: 0,
+        // gzip: zlib.gzipSync(contents).byteLength,
+        // brotli: zlib.brotliCompressSync ? zlib.brotliCompressSync(contents).byteLength : 0,
+      },
+    };
+  }
+
+  return {
+    common: Object.assign({}, ...stats.filter((x) => x.isCommon).map(makeStatObject)),
+    direct: Object.assign({}, ...stats.filter((x) => !x.isCommon).map(makeStatObject)),
+  };
+}
+
+async function bundleWithRollup({installEntrypoints, installTargets, ...options}: BundlerOptions) {
+  const {
+    cwd,
+    alias: installAlias,
+    logger,
+    dest: destLoc,
+    namedExports,
+    externalPackage: externalPackages,
+    sourceMap,
+    env: userEnv,
+    rollup: userDefinedRollup,
+    treeshake: isTreeshake,
+    polyfillNode,
+  } = setOptionDefaults(options);
```

Comment:
Soft suggestion (that verges on bikeshedding): it’d be nice to try and separate Rollup logic from esbuild logic. I think we’d like to only use esbuild once it can do everything Rollup can. Having some separation where it makes sense might make the switchover easier.

I say that, but maybe the distinction is moot and there’s really not that much separating the two other than the final output.